projectdo

```diff
@@ -34,13 +34,17 @@ has_command() {
 # evaluates the given command while respecting $QUIET and $DRY_RUN.
 execute() {
   if [ $QUIET = false ]; then
-    echo "$1" $TOOL_ARGS
+    if [ -z "$TOOL_ARGS" ]; then
+      printf '%s' "$1"
+    else
+      printf '%s %s' "$1" "$TOOL_ARGS"
+    fi
   fi
   if [ $DRY_RUN = false ]; then
     if [ $QUIET = false ]; then
-      eval "$1" $TOOL_ARGS
+      eval "$1" "$TOOL_ARGS"
```

Comment:
I think this will pass all the arguments along to the tool as one single argument with spaces. We want to actually pass the arguments along to the tool as multiple arguments.

I think we need to preserve the `TOOL_ARGS=$@` below. I found [this SO answer](https://stackoverflow.com/a/9995322/556072) which seems imply that `$@`. We need to keep the echo line or find something equivalent. Shellcheck complains about the `eval` but maybe that is a false positive or maybe there is a better way of achieving what the original `eval` line does in a better way?

Reply:
Okay, I reverted it back to the previous state but we should probably look into that a bit more because judging from the information [here](https://www.shellcheck.net/wiki/SC2124), the behavior of `string=$@` is dependent on the shell and using `$*` would eliminate that uncertainty.

Reply:
That is a good idea. I think I'll see if I can add a test that ensures that the arguments are passed correctly, and then from there we can refactor while ensuring that the behavior is preserved :)